### ISSUE TITLE
dotnet publish for Solution Filter fails with @ in path

### ISSUE THREAD

**Issue by astaerk** [dotnet/sdk#29141]
I use ` dotnet publish` in a Jenkins job. When multiple builds run in parallel, Jenkins automatically appends an @ and a number to the path of the working directory. So for example the working directory is called like this:
`C:\Jenkins\Workspace\SomeApplication@2` 
(I can totally reproduce the error on my local machine, but Jenkins is the reason why I don't just rename the directory and go on.)

Because the application is published in multiple parts, I use a solution filter (*.slnf) to group the correct projects together.

When I now call the publish command for the solution filter in this directory, I get an error stating, that it couldn't find a  part of the path. When you have a look at the error message (see below) you will notice, that the **@2** in the path got replaced by **%402** internally in the dotnet commant. But that path obviously doesn't exist and so it can't find it.

### To Reproduce
I created a small repo to reproduce the issue:
https://github.com/astaerk/DotnetPublishErrorDemo
#### Remarks
- normally the Folder with the @ in the name is NOT part of the repo. As I wrote, it gets created by Jenkins, that then clones the code in that directory. But for the sake of the example I added that folder to the git repo.
- You can ignore all the actual code in the repo. I just called File/New/Project twice to simulate a need for a solution filter. But all that is really relevant are the publish.ps1 and the SomeWebApplication.slnf with the corresponding WebApplications.sln

#### Steps
1. clone the repo
2. execute publish.ps1 (or the dotnet publish command that is inside of it)
3. You will get the error

### Exception / Error Message
`C:\Jenkins\Workspace\SomeApplication@2\API\API.slnf : error MSB4014: Der Buildvorgang wurde aufgrund eines internen Fehlers unerwartet beendet.
C:\Jenkins\Workspace\SomeApplication@2\API\API.slnf : error MSB4014: System.IO.DirectoryNotFoundException: Could not find a part of the path 'C:\Jenkins\Workspace\SomeApplication%402\SomeApplication.sln'.
C:\Jenkins\Workspace\SomeApplication@2\API\API.slnf : error MSB4014:    at Microsoft.Win32.SafeHandles.SafeFileHandle.CreateFile(String fullPath, FileMode mode, FileAccess access, FileShare share, FileOptions options)
C:\Jenkins\Workspace\SomeApplication@2\API\API.slnf : error MSB4014:    at Microsoft.Win32.SafeHandles.SafeFileHandle.Open(String fullPath, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize)
C:\Jenkins\Workspace\SomeApplication@2\API\API.slnf : error MSB4014:    at System.IO.Strategies.OSFileStreamStrategy..ctor(String path, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize)
C:\Jenkins\Workspace\SomeApplication@2\API\API.slnf : error MSB4014:    at System.IO.Strategies.FileStreamHelpers.ChooseStrategyCore(String path, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize)
C:\Jenkins\Workspace\SomeApplication@2\API\API.slnf : error MSB4014:    at System.IO.Strategies.FileStreamHelpers.ChooseStrategy(FileStream fileStream, String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options, Int64 preallocationSize)
C:\Jenkins\Workspace\SomeApplication@2\API\API.slnf : error MSB4014:    at System.IO.File.OpenRead(String path)
C:\Jenkins\Workspace\SomeApplication@2\API\API.slnf : error MSB4014:    at Microsoft.Build.Construction.SolutionFile.GetSolutionFileAndVisualStudioMajorVersions(String solutionFile, Int32& solutionVersion, Int32& visualStudioMajorVersion)
C:\Jenkins\Workspace\SomeApplication@2\API\API.slnf : error MSB4014:    at Microsoft.Build.Execution.ProjectInstance.LoadSolutionForBuild(String projectFile, PropertyDictionary``1 globalPropertiesInstances, String toolsVersion, BuildParameters buildParameters, ILoggingService loggingService, BuildEventContext projectBuildEventContext, Boolean isExplicitlyLoaded, IReadOnlyCollection``1 targetNames, ISdkResolverService sdkResolverService, Int32 submissionId)
C:\Jenkins\Workspace\SomeApplication@2\API\API.slnf : error MSB4014:    at Microsoft.Build.Execution.BuildManager.LoadSolutionIntoConfiguration(BuildRequestConfiguration config, BuildRequest request)
C:\Jenkins\Workspace\SomeApplication@2\API\API.slnf : error MSB4014:    at Microsoft.Build.Execution.BuildManager.HandleNewRequest(Int32 node, BuildRequestBlocker blocker)
C:\Jenkins\Workspace\SomeApplication@2\API\API.slnf : error MSB4014:    at Microsoft.Build.Execution.BuildManager.<>c__DisplayClass98_0.<IssueBuildRequestForBuildSubmission>g__IssueBuildSubmissionToSchedulerImpl|1(BuildSubmission submission, Boolean allowMainThreadBuild)`

### Further technical details
Visual Studio 17.4.0

.NET SDK:
 Version:   7.0.100
 Commit:    e12b7af219

Laufzeitumgebung:
 OS Name:     Windows
 OS Version:  10.0.19044
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\7.0.100\

Host:
  Version:      7.0.0
  Architecture: x64
  Commit:       d099f075e4

.NET SDKs installed:
  4.1.425 [C:\Program Files\dotnet\sdk]
  5.0.408 [C:\Program Files\dotnet\sdk]
  5.0.414 [C:\Program Files\dotnet\sdk]
  6.0.306 [C:\Program Files\dotnet\sdk]
  7.0.100 [C:\Program Files\dotnet\sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 3.1.31 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 5.0.15 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 5.0.17 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 6.0.11 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 7.0.0 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 2.1.30 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.31 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 5.0.15 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 5.0.17 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.11 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 7.0.0 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 3.1.31 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 5.0.15 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 5.0.17 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 6.0.11 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 7.0.0 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]

Other architectures found:
  x86   [C:\Program Files (x86)\dotnet]
    registered at [HKLM\SOFTWARE\dotnet\Setup\InstalledVersions\x86\InstallLocation]

Environment variables:
  Not set

global.json file:
  Not found

Learn more:
  https://aka.ms/dotnet/info

Download .NET:
  https://aka.ms/dotnet/download

**Comment by chrisheil** [comment]
I am seeing the same issue as well with .NET 8
